dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java

```diff
@@ -54,4 +63,109 @@ public CategoryView toCategoryView(final Category category, final User user) {
+        try{
+            csvreader = new CsvReader(br);
+            csvreader.setSafetySwitch(false);
+            csvreader.readHeaders();
+            String[] csvLine;
+
+            while (csvreader.readRecord()) {
+                csvLine = csvreader.getValues();
+                try {
+                    addOrUpdateCategory(user, true, contextInode, csvLine[0], csvLine[2],
+                            csvLine[1], null, csvLine[3], merge);
+
+                } catch (Exception e) {
+                    Logger.error(this,
+                            "Error trying to save/update the categories csv row: name=" + csvLine[0]
+                                    + ", variable=" + csvLine[2] + ", key=" + csvLine[1] + ", sort="
+                                    + csvLine[3], e);
+                }
+            }
+        }
+        finally {
+            csvreader.close();
+            br.close();
+        }
+    }
+
+    private Integer addOrUpdateCategory(final User user, final Boolean isSave, final String inode, final String name, final String var, final String key, final String keywords, final String sort, final boolean isMerge)
+            throws Exception {
+
+        Category parent = null;
+        Category cat = new Category();
+        cat.setCategoryName(name);
+        cat.setKey(key);
+        cat.setCategoryVelocityVarName(var);
+        cat.setSortOrder(sort);
+        cat.setKeywords(keywords);
+
+        if(UtilMethods.isSet(inode)){
+            if(!isSave){//edit
+                cat.setInode(inode);
+                final Category finalCat = cat;//this is to be able to use the try.of
+                parent = Try.of(()->categoryAPI.getParents(finalCat,user,false).get(0)).getOrNull();
+            }else{//save
+                parent = categoryAPI.find(inode, user, false);
+            }
+        }
+
+        setVelocityVarName(cat, var, name);
+
+        if(isMerge) { // add/edit
+
+            if(isSave) { // Importing
+                if(UtilMethods.isSet(key)) {
+                    cat = categoryAPI.findByKey(key, user, false);
+                    if(cat==null) {
+                        cat = new Category();
+                        cat.setKey(key);
+                    }
+
+                    cat.setCategoryName(name);
+                    setVelocityVarName(cat, var, name);
+                    cat.setSortOrder(sort);
+                }
+            } else { // Editing
+                cat = categoryAPI.find(inode, user, false);
+                cat.setCategoryName(name);
+                setVelocityVarName(cat, var, name);
+                cat.setKeywords(keywords);
+                cat.setKey(key);
+            }
+
+        } else { // replace
+            cat.setCategoryName(name);
+            setVelocityVarName(cat, var, name);
+            cat.setSortOrder(sort);
+            cat.setKey(key);
+        }
+
+        try {
+            categoryAPI.save(parent, cat, user, false);
```

Comment:
use vavr lib such as

Try.run (()-> categoryAPI.save(parent, cat, user, false));

Reply:
we need to make parent and category as final otherwise we cannot change to 
Try.run (()-> categoryAPI.save(parent, cat, user, false));

and parent is being assigned later like this 


 if(UtilMethods.isSet(inode)){
            if(!isSave){//edit
                category.setInode(inode);
                final Category finalCat = category;//this is to be able to use the try.of
                parent = Try.of(()->categoryAPI.getParents(finalCat,user,false).get(0)).getOrNull();
            }else{//save
                parent = categoryAPI.find(inode, user, false);
            }
        }